ros_ws/src/crazyswarm/src/mocap_helper.cpp

```diff
@@ -29,98 +15,42 @@ int main(int argc, char **argv)
   std::string motionCaptureType;
   nl.param<std::string>("motion_capture_type", motionCaptureType, "vicon");
 
-  // Make a new client
-  libmotioncapture::MotionCapture* mocap = nullptr;
-  if (false)
-  {
-  }
-#ifdef ENABLE_VICON
-  else if (motionCaptureType == "vicon")
-  {
-    std::string hostName;
-    nl.getParam("vicon_host_name", hostName);
-    mocap = new libmotioncapture::MotionCaptureVicon(hostName,
-      /*enableObjects*/ true,
-      /*enablePointcloud*/ true);
-  }
-#endif
-#ifdef ENABLE_OPTITRACK
-  else if (motionCaptureType == "optitrack")
-  {
-    std::string hostName;
-    nl.getParam("optitrack_host_name", hostName);
-    mocap = new libmotioncapture::MotionCaptureOptitrack(hostName);
-  }
-#endif
-#ifdef ENABLE_PHASESPACE
-  else if (motionCaptureType == "phasespace")
-  {
-    std::string ip;
-    int numMarkers;
-    nl.getParam("phasespace_ip", ip);
-    nl.getParam("phasespace_num_markers", numMarkers);
-    std::map<size_t, std::pair<int, int> > cfs;
-    cfs[231] = std::make_pair<int, int>(10, 11);
-    mocap = new libmotioncapture::MotionCapturePhasespace(ip, numMarkers, cfs);
-  }
-#endif
-#ifdef ENABLE_QUALISYS
-  else if (motionCaptureType == "qualisys")
-  {
-    std::string hostname;
-    int port;
-    nl.getParam("qualisys_host_name", hostname);
-    nl.getParam("qualisys_base_port", port);
-    mocap = new libmotioncapture::MotionCaptureQualisys(hostname, port,
-      /*enableObjects*/ true,
-      /*enablePointcloud*/ true);
-  }
-#endif
-#ifdef ENABLE_VRPN
-  else if (motionCaptureType == "vrpn")
-  {
-    std::string hostname;
-    int port;
-    nl.getParam("vrpn_host_name", hostname);
-    mocap = new libmotioncapture::MotionCaptureVrpn(hostname);
-  }
-#endif
-  else {
+  std::map<std::string, std::string> cfg;
+  std::string hostname;
+  nl.getParam("motion_capture_host_name", hostname);
+  cfg["hostname"] = hostname;
+  libmotioncapture::MotionCapture *mocap = libmotioncapture::MotionCapture::connect(motionCaptureType, cfg);
```

Comment:
Is this memory `libmotioncapture::MotionCapture *mocap` ever freed? Could this be a `std::unique_pointer`?